core/src/main/java/hivemall/fm/FactorizationMachineModel.java

```diff
@@ -399,9 +399,8 @@ public void initRandom(int factor, long seed) {
     protected static final void uniformFill(final float[] a, final Random rand,
             final float maxInitValue) {
         final int len = a.length;
-        final float basev = maxInitValue / len;
         for (int i = 0; i < len; i++) {
-            float v = rand.nextFloat() * basev;
+            float v = rand.nextFloat() * maxInitValue;
```

Comment:
While this modified `random` initialization is not used for classification (and only for regression), your evaluation is only for classification. 

This, it's doubtful that this change contributed for improving accuracy.